src/Tasks/Microsoft.NET.Build.Tasks/GetPackageDirectory.cs

```diff
@@ -12,55 +12,53 @@ namespace Microsoft.NET.Build.Tasks
     //  but not PackageDirectory metadata specified
     public class GetPackageDirectory : TaskBase
     {
-        public ITaskItem[] Items { get; set; }
+        public ITaskItem[] Items { get; set; } = Array.Empty<ITaskItem>();
 
-        [Required]
-        public string ProjectPath { get; set; }
-
-        [Required]
-        public string [] PackageFolders { get; set; }
+        public string[] PackageFolders { get; set; } = Array.Empty<string>();
 
         [Output]
         public ITaskItem[] Output { get; set; }
 
         protected override void ExecuteCore()
         {
-            if (Items != null)
+            if (Items.Length == 0 || PackageFolders.Length == 0)
             {
-                NuGetPackageResolver packageResolver = NuGetPackageResolver.CreateResolver(PackageFolders, ProjectPath);
+                Output = Items;
```

Comment:
It seems like this is avoiding a call to `Array.Empty<T>()`, but makes the code less clear.  Unless we have reason to believe that the method call will have a perf impact, I'd go with clarity and make the call to `Array.Empty<T>()` here.

Reply:
For empty package folders, array.empty is incorrect. I could split the cases if you like.

Reply:
Never mind, I was confused.